src/lib/DataConnectStore.js

```diff
@@ -145,6 +145,10 @@ export default class DataConnectStore {
     connector.accounts[accountIdx].auth.password = values.password
 
     return accounts.update(cozy.client, previousAccount, connector.accounts[accountIdx])
+    .then(newAccountData => {
+      // Updates the account with the serveur answer (mostly updates the _rev value)
+      connector.accounts[accountIdx] = newAccountData
```

Comment:
We should call `updateConnector(connector)` right after to keep a trace of the new account into the store.